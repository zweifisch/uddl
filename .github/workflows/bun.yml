name: Bun

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v4
    - name: Setup Bun
      uses: oven-sh/setup-bun@v2.0.1
    - name: Install Dependencies
      run: bun install --frozen-lockfile
    - name: Run tests
      run: bun test
