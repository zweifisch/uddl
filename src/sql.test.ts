import {test, expect} from 'bun:test'
import { toSQL } from './'

test('toSQL', () => {

  const input = `
User {
  id(primary_key auto_increment): int,
  name?: text,
  email(unique): text,
  gender(default:0): int,
}
`
  expect(toSQL(input)).toMatchSnapshot()
})