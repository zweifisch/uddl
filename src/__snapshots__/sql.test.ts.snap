// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`toSQL 1`] = `
"CREATE TABLE user (
  id INTEGER PRIMARY KEY AUTOINCREMENT,
  name TEXT,
  email TEXT UNIQUE NOT NULL,
  gender INTEGER DEFAULT 0 NOT NULL
);"
`;
